matsim/src/main/java/ch/sbb/matsim/routing/pt/raptor/SwissRailRaptorData.java

```diff
@@ -42,547 +42,603 @@
+			for (TransitStopFacility toStop : nearbyStops) {
+				int[] toRouteStopIndices = routeStopsPerStopFacility.get(toStop);
+				double beelineDistance = CoordUtils.calcEuclideanDistance(fromCoord, toStop.getCoord());
+				double transferTime = beelineDistance / beelineWalkSpeed;
+				if (transferTime < minimalTransferTime) {
+					transferTime = minimalTransferTime;
+				}
+
+				transferTime = mtt.get(fromStop.getId(), toStop.getId(), transferTime);
+
+				final double fixedTransferTime = transferTime; // variables must be effective final to be used in
+																// lambdas (below)
+
+				for (int fromRouteStopIndex : fromRouteStopIndices) {
+					RRouteStop fromRouteStop = routeStops[fromRouteStopIndex];
+					stopTransfers.clear();
+					for (int toRouteStopIndex : toRouteStopIndices) {
+						RRouteStop toRouteStop = routeStops[toRouteStopIndex];
+						if (isUsefulTransfer(fromRouteStop, toRouteStop, maxBeelineWalkConnectionDistance,
+								config.getOptimization())) {
+							RTransfer newTransfer = new RTransfer(fromRouteStopIndex, toRouteStopIndex,
+									fixedTransferTime, beelineDistance * beelineDistanceFactor);
+							stopTransfers.add(newTransfer);
+						}
+					}
+					RTransfer[] newTransfers = stopTransfers.toArray(new RTransfer[0]);
+					transfers.compute(fromRouteStopIndex, (routeStopIndex, currentTransfers) -> {
+						if (currentTransfers == null) {
+							return newTransfers;
+						}
+						RTransfer[] tmp = new RTransfer[currentTransfers.length + newTransfers.length];
+						System.arraycopy(currentTransfers, 0, tmp, 0, currentTransfers.length);
+						System.arraycopy(newTransfers, 0, tmp, currentTransfers.length, newTransfers.length);
+						return tmp;
+					});
+				}
+			}
+		}
+		return transfers;
+	}
+
+	private static boolean isUsefulTransfer(RRouteStop fromRouteStop, RRouteStop toRouteStop,
+			double maxBeelineWalkConnectionDistance, RaptorStaticConfig.RaptorOptimization optimization) {
+		if (fromRouteStop == toRouteStop) {
+			return false;
+		}
+		// there is no use to transfer away from the first stop in a route
+		if (isFirstStopInRoute(fromRouteStop)) {
+			return false;
+		}
+		// there is no use to transfer to the last stop in a route, we can't go anywhere
+		// from there
+		if (isLastStopInRoute(toRouteStop)) {
+			return false;
+		}
+		// if the first departure at fromRouteStop arrives after the last departure at
+		// toRouteStop,
+		// we'll never get any connection here
+		if (hasNoPossibleDeparture(fromRouteStop, toRouteStop)) {
+			return false;
+		}
+		// if the stop facilities are different, and the destination stop is part
+		// of the current route, it does not make sense to transfer here
+		if (toStopIsPartOfRouteButNotSame(fromRouteStop, toRouteStop)) {
+			return false;
+		}
+		// assuming vehicles serving the exact same stop sequence do not overtake each
+		// other,
+		// it does not make sense to transfer to another route that serves the exact
+		// same upcoming stops
```

Comment:
seriously, do I have to look over all thees changes? It's not even nicer formatted with single words on a line. How should that "auto-format" improve legibility?

Reply:
Seriously, sorry. will make another commit with only the change there. The formatting came on its own...

Reply:
thanks, it looks now much more concise!